src/refda/exec_item.h

```diff
@@ -46,20 +55,32 @@ typedef struct
      */
     cace_amm_lookup_t deref;
 
-    /** Indicator if this item is waiting on some external event to finish.
-     * While true this item cannot be executed and will not yet have a
-     * valid #result.
-     */
-    atomic_bool waiting;
-
     /** Storage for an optional result value.
      * This is initialized as undefined and may be set to any other value
      * to indicate that a result is produced.
      */
     cace_ari_t result;
 
+    /**
+     * Indicator if this item is
+     * 1) yet to be completed or paused
+     * 2) waiting on some external event to complete
+     * 3) completed execution but not finished
+     * While waiting this item cannot be executed and will not yet have a
+     * valid #result.
+     *
+     */
+    atomic_int execution_stage;
```

Comment:
Annoying doc feedback:
State enum descriptions should be moved up to the enum declarations, and then this doc be updated to mention something like "Values are one of the ::refda_exec_item_status_t enumerations".

Reply:
Fixed up docstrings